test/dynamo/test_misc.py

```diff
@@ -5353,6 +5353,16 @@ def test_hash_getitem_slice(self):
         self.assertTrue(s == s2)
         self.assertTrue(s != s3)
 
+    def test_inline_dict_function(self):
+        def _result_type_dict(dtype):
+            return {bool: torch.float32}[dtype]
+
```

Comment:
It's better we can add another scenario where the dict is passed in as an argument:
```
@torch.compile()
def fn(d, x):
    if d[x] is torch.float32:
        return d[x]
    else:
        ...

dd = {bool: torch.float32}
fn(dd, bool)
```

You can refer this PR: https://github.com/pytorch/pytorch/pull/93026

Reply:
ok will do

Reply:
done